### PR TITLE
docs/adr: initialise ADR directory

### PR DESCRIPTION
**Pull request by renaynay**
This PR initialises an ADR directory for ADRs specific to the `Celestia-Node` implementation.

### CODE REVIEW

File: docs/adr/adr-template.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+>
+> - Are there any privacy considerations?
+>
+> - How will the changes be tested?
+>
+> - If the change is large, how will the changes be broken up for ease of review?
+>
+> - Will these changes require a breaking (major) release?
+>
+> - Does this change require coordination with the Celestia fork of the SDK or celestia-app?
```
Comment (liamsi) [inline docs/adr/adr-template.md:50]:
```suggestion
> - Does this change require coordination with the Celestia fork of the SDK, celestia-app/-core, or any other celestiaorg repository?
```

**Review by liamsi — DISMISSED** [review]
Minor suggestion.

**Review by Wondertan — APPROVED** [review]
Let's have ADRs in here, but it also makes sense to have that outside of the repo: https://github.com/celestiaorg/celestia-node/issues/16

File: docs/adr/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+new. The spec is much more compressed and streamlined summary of everything as
+it stands today.
+
+If recorded decisions turned out to be lacking, convene a discussion, record the new decisions here, and then modify the code to match.
+
+Note the context/background should be written in the present tense.
+
+To start a new ADR, you can use this template: [adr-template.md](./adr-template.md)
+
+### Table of Contents:
```
Comment (adlerjohn) [inline docs/adr/README.md:32]: Is there a reason for this?

Comment (liamsi) [reply docs/adr/README.md:32]: I assume this will contain a list of all ADRs contained in this repo.

**Review by liamsi — APPROVED** [review]

File: docs/adr/README.md (hunk 2 above)
Comment (renaynay) [reply docs/adr/README.md:32]: Yes.

**Pull request merged**